samples/bluetooth/broadcast_audio_sink/src/main.c

```diff
@@ -122,60 +244,101 @@ static int lc3_enable(const struct bt_audio_codec_cfg *codec_cfg)
+#if defined(CONFIG_USB_DEVICE_AUDIO)
+/* Duplicate the audio from one channel and put it in both channels */
+void mix_mono_to_stereo(void)
 {
-	int err = 0;
-	int offset = 0;
-	uint8_t *buf_data;
-	struct net_buf *ptr_net_buf;
-	int octets_per_frame;
-	struct broadcast_sink_stream *sink_stream = CONTAINER_OF(
-		k_work_delayable_from_work(work), struct broadcast_sink_stream, lc3_decode_work);
+	int16_t usb_audio_data[MAX_NUM_SAMPLES_MONO] = {0};
+	int16_t usb_audio_data_stereo[MAX_NUM_SAMPLES_STEREO] = {0};
+	uint32_t size;
+
+	size = ring_buf_get(&streams[0].audio_ring_buf, (uint8_t *)usb_audio_data,
+			    MAX_NUM_SAMPLES_STEREO);
+	if (size != MAX_NUM_SAMPLES_STEREO) {
+		memset(&((uint8_t *)usb_audio_data)[size], 0, MAX_NUM_SAMPLES_STEREO - size);
+	}
 
-	k_mutex_lock(&sink_stream->lc3_decoder_mutex, K_FOREVER);
+	/* Interleave the channel sample */
+	for (size_t i = 0U; i < MAX_NUM_SAMPLES_MONO; i++) {
+		usb_audio_data_stereo[i*2] = usb_audio_data[i];
+		usb_audio_data_stereo[i*2+1] = usb_audio_data[i];
+	}
 
-	if (sink_stream->in_buf == NULL) {
-		printk("buf data is NULL, nothing to be docoded\n");
-		k_mutex_unlock(&sink_stream->lc3_decoder_mutex);
-		return;
+	size = ring_buf_put(&ring_buf_usb, (uint8_t *)usb_audio_data_stereo,
+			    BROADCAST_STEREO_SAMPLE_SIZE);
+	if (size != BROADCAST_STEREO_SAMPLE_SIZE) {
+		static int rb_put_failures;
+
+		rb_put_failures++;
+		if (rb_put_failures % 1000 == 0) {
+			printk("%s: Failure to add to ring buffer %d, %u\n", __func__,
+			       rb_put_failures, size);
+		}
 	}
+}
 
-	ptr_net_buf = net_buf_ref(sink_stream->in_buf);
-	net_buf_unref(sink_stream->in_buf);
-	sink_stream->in_buf = NULL;
-	k_mutex_unlock(&sink_stream->lc3_decoder_mutex);
+/* USB consumer callback, called every 1ms, consumes data from ring-buffer */
+static void data_request(const struct device *dev)
+{
+	static struct net_buf *pcm_buf;
+	size_t in_frame_size;
+	int err;
+	uint32_t size;
+	void *out;
+	int16_t usb_audio_data[USB_STEREO_SAMPLE_SIZE] = {0};
 
-	buf_data = ptr_net_buf->data;
-	octets_per_frame = ptr_net_buf->len / frames_per_sdu;
+	in_frame_size =	usb_audio_get_in_frame_size(dev);
 
-	for (int i = 0; i < frames_per_sdu; i++) {
-		err = lc3_decode(lc3_decoder, buf_data + offset, octets_per_frame,
-						 LC3_PCM_FORMAT_S16, audio_buf, 1);
+	size = ring_buf_get(&ring_buf_usb, (uint8_t *)usb_audio_data, USB_STEREO_SAMPLE_SIZE);
+	if (size != USB_STEREO_SAMPLE_SIZE) {
+		memset(&((uint8_t *)usb_audio_data)[size], 0, USB_STEREO_SAMPLE_SIZE);
+	}
+	pcm_buf = net_buf_alloc(&usb_tx_buf_pool, K_NO_WAIT);
+	if (pcm_buf == NULL) {
+		printk("Couldnt allocate pcm_buf\n");
+		return;
+	}
 
-		if (err == 1) {
-			printk("  decoder performed PLC\n");
-		} else if (err < 0) {
-			printk("  decoder failed - wrong parameters?\n");
-		}
+	out = net_buf_add(pcm_buf, USB_STEREO_SAMPLE_SIZE);
+	memset(out, 0, USB_STEREO_SAMPLE_SIZE);
+	memcpy(out, usb_audio_data, USB_STEREO_SAMPLE_SIZE);
```

Comment:
Why do we need the `memset` here? Aren't we writing the same size to the same pointer twice?

Reply:
Yes, this is a mistake